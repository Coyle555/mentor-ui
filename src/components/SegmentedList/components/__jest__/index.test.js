jest.mock('react-spring/renderprops', () => {
	return props => ({
		config: {},
		Spring: (
			<div>
				{JSON.stringify(props)}
				{ props.children() }
			</div>
		)
	});
});

import React from 'react';
import { SegmentedList } from '../index';
import renderer from 'react-test-renderer';

test('Default render of segmented list', () => {
	const tree = renderer.create(<SegmentedList />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Segmented list with items', () => {
	const tree = renderer.create(
		<SegmentedList items={['Foo', 'Bar', 'Baz']} />
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Segmented list with title', () => {
	const tree = renderer.create(<SegmentedList title="Test" />).toJSON();

	expect(tree).toMatchSnapshot();
});
