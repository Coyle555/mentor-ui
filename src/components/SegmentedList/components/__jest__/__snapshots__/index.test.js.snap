// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render of segmented list 1`] = `
<div
  className="mui-segmented-list"
>
  <ul
    className="mui-segmented-list-ul"
  />
</div>
`;

exports[`Segmented list with items 1`] = `
<div
  className="mui-segmented-list"
>
  <ul
    className="mui-segmented-list-ul"
  >
    <li
      className="list-item"
    >
      Foo
    </li>
    <li
      className="list-item"
    >
      Bar
    </li>
    <li
      className="list-item"
    >
      Baz
    </li>
  </ul>
</div>
`;

exports[`Segmented list with title 1`] = `
<div
  className="mui-segmented-list"
>
  <div
    className="title-container"
  >
    <h2
      className="title"
    >
      Test
    </h2>
  </div>
  <ul
    className="mui-segmented-list-ul"
  />
</div>
`;
